biz.aQute.launchpad/src/aQute/launchpad/BundleSpecBuilder.java

```diff
@@ -871,6 +871,28 @@ default BundleSpecBuilder addResource(Class<?> class1) {
 		return this;
 	}
 
+	/**
+	 * Actually add the specified class into the bundle. This contrasts with
+	 * {@link #addResource(Class)}, which imports the specified class via OSGi
+	 * (typically from the system classloader). Classes added using this method
+	 * will be loaded from within the framework, rather than by the system
+	 * framework.
+	 * 
+	 * @param class1 the class to add as a resource
+	 */
+	default BundleSpecBuilder addResourceWithCopy(Class<?> class1) {
+		String className = class1.getName();
+		String path = className.replace('.', '/') + ".class";
+		URL url = BundleSpecBuilder.class.getClassLoader()
```

Comment:
Shouldn't this use the specified class' class loader to load the resource? For example, `URL url = class1.getClassLoader().getResource(path);`? Or even `URL url = class1.getResource(class1.getSimpleName()+".class");` since the class file resource is in the package of the class.

Reply:
I think I tried this and it didn't work. I'll try again.

Reply:
Ok:
- `getSimpleName()` doesn't work with nested classes. Eg, for `my.pkg.Clazz.InnerClass` it simply gives you `InnerClass`, but the class file on disk will be `Clazz$InnerClass.class`.
- Using class1's classloader works. I can't remember why I used `BundleSpecBuilder`'s classloader, but I think it must have been either a misguided micro-optimisation, or else a miscalculation as to how class visibility issues might play out. I'll change it to use class1's classloader.